tests/integration/test_records.py

```diff
@@ -367,3 +369,54 @@ def test_records_files_attached_correctly(app):
     record.commit()
 
     assert 'fulltext.pdf' in record.files
+
+
+@patch('inspirehep.modules.records.utils.open', mock_open())
+def test_records_files_are_populated_correctly_for_local_paths(app):
+    record_json = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        'document_type': [
+            'article',
+        ],
+        'titles': [
+            {'title': 'foo'},
+        ],
+        '_collections': ['Literature'],
+        'documents': [{
+            'key': 'arXiv:1710.01187.pdf',
+            'url': '/afs/cern.ch/project/inspire/PROD/var/data/files/g151/3037619/content.pdf;1',
+        }]  # record/1628455/export/xme
+    }
+
+    record = InspireRecord.create(record_json)
+    update_documents_and_figures_for_record(record)
+
+    assert 'arXiv:1710.01187.pdf' in record.files
+
+
+def test_records_files_are_populated_correctly_for_remote_paths(app):
+        record_json = {
+            '$schema': 'http://localhost:5000/schemas/records/hep.json',
+            'document_type': [
+                'article',
+            ],
+            'titles': [
+                {'title': 'foo'},
+            ],
+            '_collections': ['Literature'],
+            'documents': [{
+                'key': 'Fulltext.pdf',
+                'url': 'http://www.mdpi.com/2218-1997/3/1/24/pdf',
+            }]  # DESY harvest
+        }
+
+        record = InspireRecord.create(record_json, id_=None)
```

Comment:
Do you need `id_=None` here and not above?

Reply:
no this a leftover.